packages/wallet/src/W3mFrameProvider.ts

```diff
@@ -0,0 +1,332 @@
+import { W3mFrame } from './W3mFrame.js'
+import type { W3mFrameTypes } from './W3mFrameTypes.js'
+import { W3mFrameConstants } from './W3mFrameConstants.js'
+import { W3mFrameStorage } from './W3mFrameStorage.js'
+
+// -- Types -----------------------------------------------------------
+type Resolver<T> = { resolve: (value: T) => void; reject: (reason?: unknown) => void } | undefined
+type ConnectEmailResolver = Resolver<W3mFrameTypes.Responses['FrameConnectEmailResponse']>
+type ConnectDeviceResolver = Resolver<undefined>
+type ConnectOtpResolver = Resolver<undefined>
+type ConnectResolver = Resolver<W3mFrameTypes.Responses['FrameGetUserResponse']>
+type DisconnectResolver = Resolver<undefined>
+type IsConnectedResolver = Resolver<W3mFrameTypes.Responses['FrameIsConnectedResponse']>
+type GetChainIdResolver = Resolver<W3mFrameTypes.Responses['FrameGetChainIdResponse']>
+type SwitchChainResolver = Resolver<undefined>
+type RpcRequestResolver = Resolver<W3mFrameTypes.RPCResponse>
+
+// -- Provider --------------------------------------------------------
+export class W3mFrameProvider {
+  private w3mFrame: W3mFrame
+
+  private connectEmailResolver: ConnectEmailResolver = undefined
+
+  private connectDeviceResolver: ConnectDeviceResolver = undefined
+
+  private connectOtpResolver: ConnectOtpResolver | undefined = undefined
+
+  private connectResolver: ConnectResolver = undefined
+
+  private disconnectResolver: DisconnectResolver = undefined
+
+  private isConnectedResolver: IsConnectedResolver = undefined
+
+  private getChainIdResolver: GetChainIdResolver = undefined
+
+  private switchChainResolver: SwitchChainResolver = undefined
+
+  private rpcRequestResolver: RpcRequestResolver = undefined
+
+  public constructor(projectId: string) {
+    this.w3mFrame = new W3mFrame(projectId, true)
+    this.w3mFrame.events.onFrameEvent(event => {
+      // eslint-disable-next-line no-console
+      console.log('💻 received', event)
```

Comment:
Should we remove this? @0xAsimetriq

Reply:
Would like to keep these in until we hit beta. These logs are aligned with similar ones on secure.web3modal.com iframe, to help with debugging of message/event flows.